### PR TITLE
BTSDK: removed exit sys call for ConnectionRefusedError in _get_substrate

### PR DESCRIPTION
**Pull request by basfroman**

### CODE REVIEW

File: bittensor/core/subtensor.py [hunk 1]
```diff
@@ -223,6 +222,11 @@ def _get_substrate(self):
...
                 )
 
+            try:
+                self.substrate.websocket.settimeout(self._connection_timeout)
+            except (AttributeError, TypeError, socket.error, OSError) as e:
+                logging.warning(f"Error setting timeout: {e}")
+
         except ConnectionRefusedError:
             logging.error(
                 f"Could not connect to {self.network} network with {self.chain_endpoint} chain endpoint. Exiting...",
```
Comment (thewhaleking) [inline bittensor/core/subtensor.py]: You're gonna want to change this line since we're no longer exiting.

**Review by thewhaleking — CHANGES_REQUESTED** [review]
Shouldn't we throw an Exception?

**Comment by basfroman** [comment]
> Shouldn't we throw an Exception?

I don't think so. Just warning

File: bittensor/core/subtensor.py (hunk 1 above)
Comment (basfroman) [reply bittensor/core/subtensor.py]: > You're gonna want to change this line since we're no longer exiting.

log message is changed

**Comment by thewhaleking** [comment]
> > Shouldn't we throw an Exception?
> 
> I don't think so. Just warning

Doesn't that just pass the buck down the line, and will bring a new exception when a request is attempted on a closed connection?

**Comment by basfroman** [comment]
> Doesn't that just pass the buck down the line, and will bring a new exception when a request is attempted on a closed connection?

No, if the connection data was passed correctly, then even with a missing connection, the reconnection logic will create a new connection and set the timeout again. Otherwise, the user will receive a warning again.

**Review by garrett-opentensor — APPROVED** [review]

**Review by thewhaleking — APPROVED** [review]

**Pull request merged**